src/Symfony/Component/Translation/Loader/YamlFileLoader.php

```diff
@@ -40,11 +40,15 @@ protected function loadResource($resource)
         }
 
         try {
-            $messages = $this->yamlParser->parseFile($resource, Yaml::PARSE_CONSTANT);
+            $messages = $this->yamlParser->parseFile($resource, Yaml::PARSE_CONSTANT) ?? [];
         } catch (ParseException $e) {
             throw new InvalidResourceException(sprintf('Error parsing YAML, invalid file "%s"', $resource), 0, $e);
         }
 
+        if (!\is_array($messages)) {
+            throw new InvalidResourceException(sprintf('Unable to load file "%s".', $resource));
+        }
```

Comment:
`Symfony\Component\Translation\Loader\FileLoader::load()` performs the same check on the result of `loadResource()`. We could either duplicate the check here or change the return type to `mixed`.